x-pack/plugins/cases/server/routes/api/cases/sub_case/delete_sub_cases.ts

```diff
@@ -38,56 +26,10 @@ export function initDeleteSubCasesApi({
           includedHiddenTypes: SAVED_OBJECT_TYPES,
         });
 
-        const [comments, subCases] = await Promise.all([
-          caseService.getAllSubCaseComments({ soClient, id: request.query.ids }),
-          caseService.getSubCases({ soClient, ids: request.query.ids }),
-        ]);
+        const user = caseService.getUser({ request });
 
-        const subCaseErrors = subCases.saved_objects.filter(
-          (subCase) => subCase.error !== undefined
-        );
-
-        if (subCaseErrors.length > 0) {
-          throw Boom.notFound(
-            `These sub cases ${subCaseErrors
-              .map((c) => c.id)
-              .join(', ')} do not exist. Please check you have the correct ids.`
-          );
-        }
-
-        const subCaseIDToParentID = subCases.saved_objects.reduce((acc, subCase) => {
-          const parentID = subCase.references.find((ref) => ref.type === CASE_SAVED_OBJECT);
-          acc.set(subCase.id, parentID?.id);
-          return acc;
-        }, new Map<string, string | undefined>());
-
-        await Promise.all(
-          comments.saved_objects.map((comment) =>
-            attachmentService.delete({ soClient, attachmentId: comment.id })
-          )
-        );
-
-        await Promise.all(request.query.ids.map((id) => caseService.deleteSubCase(soClient, id)));
-
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        const { username, full_name, email } = await caseService.getUser({ request });
-        const deleteDate = new Date().toISOString();
-
-        await userActionService.bulkCreate({
-          soClient,
-          actions: request.query.ids.map((id) =>
-            buildCaseUserActionItem({
-              action: 'delete',
-              actionAt: deleteDate,
-              actionBy: { username, full_name, email },
-              // if for some reason the sub case didn't have a reference to its parent, we'll still log a user action
-              // but we won't have the case ID
-              caseId: subCaseIDToParentID.get(id) ?? '',
-              subCaseId: id,
-              fields: ['sub_case', 'comment', 'status'],
-            })
-          ),
-        });
+        const client = await context.cases.getCasesClient();
+        await client.subCases.delete({ soClient, ids: request.query.ids, user });
```

Comment:
`soClient` and `user` shouldn't be part of the interface of the `delete` method. Consumer's of the client shouldn't be bothered with internal details. The `CasesClient` provides that information to all sub clients.

Reply:
Totally missed this, thanks for pointing it out!